### ISSUE TITLE
Enhancement: Change to a better datastructre for Timline slots

### ISSUE THREAD

**Issue by moaz-mokhtar** [tijlleenders/ZinZen-scheduler#314]
Current type of `Timeline.Slots` is `BTreeSet`, and need to think about a better data structure like `BTreeMap` or better for it.
```rust
pub struct Timeline {
    pub slots: BTreeSet<Slot>,
}
```

The main purposes for choosing `BTreeSet` that time are built-in avoid duplication, built-in sorting, no need to keys at the moment.

**Comment by tijlleenders** [comment]
Blocked by #210

**Comment by tijlleenders** [comment]
No longer relevant with new scheduler approach from #411
